src/orion/core/io/orion_cmdline_parser.py

```diff
@@ -0,0 +1,418 @@
+# -*- coding: utf-8 -*-
+"""
+:mod:`orion.core.io.orion_cmdline_parser` -- Parse command line arguments for Orion
+===================================================================================
+
+.. module:: orion_cmdline_parser
+   :platform: Unix
+   :synopsis: Parsing and building of the command line input for using script
+
+Simplify the parsing of a command line by storing every values inside an `OrderedDict`
+mapping the name of the argument to its value as a key-pair relation. Positional arguments
+are stored in the format `_X` where `X` represent the index of that argument inside
+the command line.
+
+CmdlineParser provides an interface to parse command line arguments from input but also
+utility functions to build it again as a list or an already formatted string.
+"""
+
+from collections import defaultdict, OrderedDict
+import copy
+import re
+
+from orion.core.io.cmdline_parser import CmdlineParser
+from orion.core.io.convert import infer_converter_from_file_type
+
+
+def _is_nonprior_wave(arg):
+    return arg.startswith('/') or arg == ''
+
+
+class OrionCmdlineParser():
+    """Python interface commandline parser
+
+    The `OrionCmdlineParser` is used as a way to obtain the parsing process of Orion
+    through a Python interface. It provides different dictionaries containing information
+    for different parts of the parsing process. It also exposes methods to retrieve the
+    command line with formatted values for parameters.
+
+    Parameters
+    ----------
+    config_prefix : str, optional
+        Prefix for the configuration file used by the parser to identify it.
+
+    Attributes
+    ----------
+    parser : CmdlineParser
+        Parser that will be used to parse the commandline.
+    priors_only : OrderedDict
+        An OrderedDict containing only the priors inside the commandline.
+    file_config : OrderedDict
+        An OrderedDict obtained by parsing the config file, if one was found.
+    augmented_config : OrderedDict
```

Comment:
For reference, we agreed offline to modify those to
`cmd_priors`
`file_priors`
`priors`